In [1]:
from scipy import stats
import pandas as pd
import numpy as np
import math as mt
import scipy.stats as stats
from bioinfokit.analys import stat
from scipy.stats import chi2_contingency

# Question 1

#### A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.

#### Answer: As per the problem statement above we will be using 2 sample 2 tail hypothesis test here as we have 2 samples provided in the problem statement.

#### Step 1 : Define the hypothesis : H0 hypothesis => There is no difference in the diameter of the cutlets of Unit A and B, Ha hypothesis => There is a significant difference in the diameter of the cutlets of Unit A and B

#### Step 2 : Since we have smaple data we will go for t-test here.

#### Step 3 : We have alpha = 0.05, if pvalue < alpha then we need to reject h0 (Null hypothesis) and if pvalue > alpha then we will fail to reject H0 means we will accept H0

In [2]:
df = pd.read_csv('Cutlets.csv')

In [3]:
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [4]:
#Calculating pvalue with t-test
stats.ttest_ind(df['Unit A'],df['Unit B'])

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.47223947245995)

#### Final Statement : as we can see above the pvalue we have got is 0.4722 which is greater then alpha valve we will fail to reject the Null hypothesis here

#### pvalue > alpha (0.4722 > 0.05)

#### Accept H0 => There is no difference in the diameter of the cutlets of Unit A and B

# Question 2

#### A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch. Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.

#### Answer : As per the problem statement above we will be using Anova hypothesis test here as we have more than 2 sample's data in the problem statement provided. Suppose the Average TAT of Lab1 is μ1, Lab2 is μ2, Lab3 is μ3 & Lab4 is μ4

#### Step 1 : Define the hypothesis : H0 hypothesis => μ1 = μ2 = μ3 = μ4, Ha hypothesis => At least 1 Lab's Average TAT is different // Not all the averages are same

#### Step 2 : Since it's Anova Hypothesis test we will go with F-ratio here.

#### Step 3 : We have alpha = 0.05, if pvalue < alpha then we need to reject h0 (Null hypothesis) and if pvalue > alpha then we will fail to reject H0 means we will accept H0

In [5]:
df1 = pd.read_csv('LabTAT.csv')

In [6]:
df1.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [7]:
#Calculating pvalue with the help of F-ratio
stats.f_oneway(df1.iloc[:,0], df1.iloc[:,1], df1.iloc[:,2], df1.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

#### Final Statement : as we can see above pvalue = 2.11 x 10 raise to -57 which is almost 0 and lesser than alpha value hence we reject H0 (Null hypothesis)

#### pvalue < alpha ( 2.11 x 10 raise to -57 < 0.05)

#### Accept Ha => At least 1 Lab's Average TAT is different // Not all the averages are same

# Question 3

#### Sales of products in four different regions is tabulated for males and females. Find if male-female buyer ratios are similar across regions.

#### Answer: As per the problem statement above we will be using Chi-square hypothesis test here, as the variables provided in the problem statement are Qualotative/Categorical.

#### Step 1 : Define the hypothesis : H0 hypothesis => The male-female buyer ratios are similar across the regions / All proportions are equal, Ha hypothesis => The male-female buyer ratios are not similar across the regions / Not all proportions are equal

#### Step 2 : We have alpha = 0.05, if pvalue < alpha then we need to reject h0 (Null hypothesis) and if pvalue > alpha then we will fail to reject H0 means we will accept H0

In [8]:
data = pd.DataFrame([
    [50, 142, 131, 70],
    [435, 1523, 1356, 750]], 
    index = ['Male', 'Female'], columns = ['East', 'West', 'North', 'South'])
data

East  West  North  South
Male      50   142    131     70
Female   435  1523   1356    750

In [9]:
# run chi-square test for independence
res = stat()
res.chisq(df=data)
print(res.summary)


Chi-squared test for independence

Test              Df    Chi-square    P-value
--------------  ----  ------------  ---------
Pearson            3       1.59595   0.660309
Log-likelihood     3       1.53541   0.674123



In [10]:
print(res.expected_df)


Expected frequency counts

        East      West     North     South
--  --------  --------  --------  --------
 0   42.7653   146.813   131.118   72.3042
 1  442.235   1518.19   1355.88   747.696



In [11]:
# using Pearson’s chi-squared statistic
# corrected for the Yates’ continuity
observed = np.array([[50, 142, 131, 70], [435, 1523, 1356, 750]])
chi_val, p_val, dof, expected =  chi2_contingency(observed)

chi_val, p_val, dof, expected

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [12]:
# without Yates’ correction for continuity
chi_val, p_val, dof, expected =  chi2_contingency(observed, correction=False)
chi_val, p_val, dof, expected

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [13]:
# for log-likelihood method 
chi_val, p_val, dof, expected =  chi2_contingency(observed, lambda_="log-likelihood")
chi_val, p_val, dof, expected

(1.5354067803116846,
 0.6741229013112389,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

#### Final Statement : as we can see above pvalue is greater than alpha hence we will fail to reject H0 (Null hypothesis) here.

#### pvalue > alpha => (0.66 > 0.05)

#### Accept H0 => The male-female buyer ratios are similar across the regions / All proportions are equal

# Question 4

#### TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain % of the customer order forms. Any error in order form renders it defective and has to be reworked before processing. The manager wants to check whether the defective % varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences

#### Answer: As per the problem statement above we will be using chi-square hypothesis test here, as the variables provided in the problem statement are Qualotative/Categorical.

#### Step 1 : Define the hypothesis : H0 hypothesis => the defective % does not vary by centre / the variables are independent Ha hypothesis => the defective % varies by centre / the variables are dependent

#### Step 2 : Here we will need to calculate t statistic value and critical value

#### Step 3 : We have alpha = 0.05 (5% significance level), if pvalue < alpha then we need to reject h0 (Null hypothesis) and if pvalue > alpha then we will fail to reject H0 means we will accept H0

In [14]:
df2 = pd.read_csv('Costomer+OrderForm.csv')

In [15]:
df2.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [16]:
df2.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [17]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 300 entries, 0 to 299
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Phillippines  300 non-null    object
 1   Indonesia     300 non-null    object
 2   Malta         300 non-null    object
 3   India         300 non-null    object
dtypes: object(4)
memory usage: 9.5+ KB


In [18]:
df2.dtypes

Phillippines    object
Indonesia       object
Malta           object
India           object
dtype: object

In [19]:
#The count of Error Free and Defective for given 4 centers
table = [[271,267,269,280],[29,33,31,20]]

In [20]:
data1 = pd.DataFrame([
    [271,267,269,280],
    [29,33,31,20]], 
    index = ['Error Free', 'Defective'], columns = ['Phillippines', 'Indonesia', 'Malta', 'India'])
data1

Phillippines  Indonesia  Malta  India
Error Free           271        267    269    280
Defective             29         33     31     20

In [21]:
# run chi-square test for independence
res = stat()
res.chisq(df=data1)
print(res.summary)


Chi-squared test for independence

Test              Df    Chi-square    P-value
--------------  ----  ------------  ---------
Pearson            3       3.85896   0.277102
Log-likelihood     3       4.08351   0.252587



In [22]:
print(res.expected_df)


Expected frequency counts

      Phillippines    Indonesia    Malta    India
--  --------------  -----------  -------  -------
 0          271.75       271.75   271.75   271.75
 1           28.25        28.25    28.25    28.25



#### Final Statement : Now if we analyse pvalue = 0.277 at 5% significance level (alpha = 0.05) we get pvlaue greater than alpha. Hence we will fail to reject H0 (Null Hypothesis)

#### pvalue > alpha (0.277 > 0.05)

#### Accept H0 => the defective % does not vary by centre / the variables are independent